instrumentation/kotlin-coroutines/src/test/kotlin/KotlinCoroutineTests.kt

```diff
@@ -138,6 +140,54 @@ class KotlinCoroutineTests(private val dispatcher: CoroutineDispatcher) {
     }
   }
 
+  fun launchConcurrentSuspendFunctions(numIters: Int) {
+    runBlocking {
+      for (i in 0 until numIters) {
+        GlobalScope.launch {
+          a(i.toLong())
+        }
+        GlobalScope.launch {
+          b(i.toLong())
+        }
+      }
+    }
+  }
+
+  suspend fun a(iter: Long) {
+    var span = tracer.spanBuilder("a").startSpan()
+    span.setAttribute("iter", iter)
+    var scope = currentContextWith(span)
+    delay(10)
+    a2(iter)
+    scope.close()
+    span.end()
+  }
+  suspend fun a2(iter: Long) {
+    var span = tracer.spanBuilder("a2").startSpan()
+    span.setAttribute("iter", iter)
+    var scope = currentContextWith(span)
+    delay(10)
```

Comment:
Why do you need scope and delay here?

Reply:
The delay is what forces the coroutine concurrency to be "pessimal" from a tracing perspective, switching frequently on the same Java thread among multiple coroutines and/or passing coroutines to other threads.  The scopes around delay are also there to stress test the system - before this instrumentation, we could easily see nonsensical traces like a2->b.